### PR TITLE
Update README/DEVELOPER files

### PR DESCRIPTION
**Pull request by mimischi**
Changes made in this Pull Request:
 - Update README and DEVELOPER files.

# To-Do
- [x] Update code example, how to mark a commit with a new version tag.

PR Checklist
------------
 - ~[ ] CHANGELOG updated (added [news fragment](https://github.com/hawkowl/towncrier#news-fragments) into changelog directory)?~
 - ~[ ] Issue raised/referenced?~

### CODE REVIEW

**Comment by mimischi** [comment]
@kain88-de The `DEVELOPER.rst` currently does not exactly mention which steps to follow, when creating a new release.

Should we try and bump the version in `mdbenchmark/__init__.py` when we also create a new CHANGELOG version with `towncrier`?

File: DEVELOPER.rst [hunk 1]
```diff
@@ -1,71 +1,96 @@
...
+Download code
+-------------
 
-.. code::
+First clone the repository to your local machine::
 
-   python setup.py develop
+    $ git clone https://github.com/bio-phys/MDBenchmark.git
+
+Create a virtual environment
```
Comment (kain88-de) [inline DEVELOPER.rst]: this is a conda environment. virtual environments are different for python. I would also link to the conda docs.

**Comment by mimischi** [comment]
@kain88-de I added some more text and also edited `CONTRIBUTING.md`. If you are fine with the current text, we can merge.

File: README.rst [hunk 2]
```diff
@@ -1,32 +1,74 @@
...
 
-If you are using conda:
+Installation on machines without root-privileges may require you to create a
+virtual environment. This can easily be done with ``conda``. The following
+command creates an environment called ``benchmark`` with Python 3.
 
 .. code::
+    $ conda create -n benchmark python=3
+    $ source activate benchmark
+    $ pip install mdbenchmark
```
Comment (kain88-de) [inline README.rst]: There is a conda package they can install as well. Why do over the pip route in addition to conda?

File: README.rst [hunk 3]
```diff
@@ -1,32 +1,74 @@
...
+
+    $ conda install -c conda-forge mdbenchmark
+
+Using with a virtual environment
+--------------------------------
 
-If you are using conda:
+Installation on machines without root-privileges may require you to create a
+virtual environment. This can easily be done with ``conda``. The following
+command creates an environment called ``benchmark`` with Python 3.
```
Comment (kain88-de) [inline README.rst]: Conda doesn’t need root privileges. Nor does pip normally. The environment handling is usually complicated for a new python user.

File: README.rst [hunk 4]
```diff
@@ -1,32 +1,74 @@
...
+.. code::
+   $ source activate benchmark
+   # mdbenchmark should now be usable!
+   $ mdbenchmark
+
+Features
+========
+
+- Generates benchmarks for GROMACS systems (NAMD to come soon: PR #29).
+- Automatically detects the queuing system on your high-performance cluster and submits jobs accordingly.
```
Comment (kain88-de) [inline README.rst]: Don’t mention a pr in the readme.

File: README.rst (hunk 3 above)
Comment (mimischi) [reply README.rst]: > Nor does pip normally.

pip regularly fails for users, as long as one doesn't pass `--user` during the install. That is what I was getting at there.

File: README.rst (hunk 2 above)
Comment (mimischi) [reply README.rst]: See #33.

File: README.rst (hunk 3 above)
Comment (kain88-de) [reply README.rst]: then mention the `--user` flag and what it does. A conda installation will work for a normal installation.

Comment (mimischi) [reply README.rst]: To make sure we're on the same page: do you mean also adding and explaining the `pip install --user` version together with the `conda install` version? Or just one of these? I don't mind which one to put, but as the `conda` method currently is not working, I'd rather wait until the `mdbenchmark` recipe is up-to-date.

File: DEVELOPER.rst [hunk 5]
```diff
@@ -1,71 +1,143 @@
...
+Make sure to have ``twine`` installed.
+
+The upload should work via::
+
+    $ twine upload dist/*
+
+6. Publishing to ``conda-forge``
+--------------------------------
 
+After the PyPI upload, update the ``conda-forge`` recipe.
```
Comment (mimischi) [inline DEVELOPER.rst]: @kain88-de How does this actually work? Can I also do this?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -1,11 +1,10 @@
...
-Changes made in this Pull Request:
- - 
+Changes made in this pull request:
+ -
 
 
 PR Checklist
 ------------
- - [ ] CHANGELOG updated (added [news fragment](https://github.com/hawkowl/towncrier#news-fragments) into changelog directory)?
+ - [ ] Updated `CHANGELOG.rst` ([more information](https://github.com/bio-phys/MDBenchmark/blob/master/DEVELOPER.rst#using-towncrier))?
```
Comment (kain88-de) [inline .github/PULL_REQUEST_TEMPLATE.md]: Maybe say `added changelog fragment in /changelog` and then the towncrier link

File: DEVELOPER.rst [hunk 7]
```diff
@@ -1,71 +1,143 @@
...
-Change version info in `mdbenchmark/__init__.py`
+Create a virtual environment
+----------------------------
 
-Updating CHANGELOG
-------------------
+If you are using ``conda``, you can easily create a `virtual environment`_::
 
-Install and run towncrier in the project directory.
+    $ conda create -n benchmark python=3
```
Comment (kain88-de) [inline DEVELOPER.rst]: `conda create -n benchmark  mdsynthesis click` 

This should install all dependencies. I don't care much which python version people use. The CI server tests all anyway.

In your current setup, the dependencies will be pulled from pip.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 6 above)
Comment (kain88-de) [reply .github/PULL_REQUEST_TEMPLATE.md]: Actually, your description in the developer guides looks nice. Be a bit more specific in the dev guides and then link to them from here. That would good.

File: DEVELOPER.rst [hunk 8]
```diff
@@ -1,71 +1,148 @@
...
---------------
+    $ git clone https://github.com/bio-phys/MDBenchmark.git
 
-Change version info in `mdbenchmark/__init__.py`
+Create a virtual environment
+----------------------------
 
-Updating CHANGELOG
-------------------
+If you are using ``conda``, you can easily create a `virtual environment`_::
```
Comment (kain88-de) [inline DEVELOPER.rst]: this is called a conda environment. virtual environments are something different in python lingo

File: DEVELOPER.rst [hunk 9]
```diff
@@ -1,71 +1,148 @@
...
+
+First make sure that your ``wheel`` package is up-to-date::
+
+    $ pip install --upgrade wheel
+
+Next we can generate a source distribution package and universal wheel::
+
+   $ python setup.py sdist bdist_wheel --universal
+
+Check that the source tarball in ``./dist/`` contains all important files!
```
Comment (kain88-de) [inline DEVELOPER.rst]: This would be the actual source code. The Readme and license file at the very minimum.

Comment (kain88-de) [reply DEVELOPER.rst]: Oh no also the CHANGELOG.  And check that the version on the `__init__.py` and changelog match. I made that mistake with the last release.

**Pull request merged**